global_test.go

```diff
@@ -0,0 +1,88 @@
+package sypl
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"testing"
+
+	"github.com/saucelabs/sypl/level"
+	"github.com/saucelabs/sypl/output"
+	"github.com/saucelabs/sypl/processor"
+	"github.com/saucelabs/sypl/safebuffer"
+	"github.com/saucelabs/sypl/shared"
+)
+
+func getBufferLogger(name string, lvl level.Level) (*Sypl, *safebuffer.Buffer) {
+	logger := NewDefault(name, lvl)
+	buffer, outputBuffer := output.SafeBuffer(level.Trace, processor.PrefixBasedOnMask(shared.DefaultTimestampFormat))
+	logger.AddOutputs(outputBuffer)
+
+	return logger, buffer
+}
+
+func TestRedirectStdLogAt(t *testing.T) {
+	initialFlags := log.Flags()
+	initialPrefix := log.Prefix()
+
+	levels := []level.Level{level.Trace, level.Debug, level.Info, level.Warn, level.Error}
+	for _, lvl := range levels {
+		runRedirectTest(t, lvl)
+	}
+	checkInitialFlags(t, initialFlags, initialPrefix)
+}
+
+func TestRedirectStdLogAtInvalid(t *testing.T) {
+	logger := NewDefault("test-invalid", level.Debug)
+	restore, err := RedirectStdLogAt(logger, level.None)
+	defer func() {
+		if restore != nil {
+			restore()
+		}
+	}()
+
+	if err == nil {
```

Comment:
I always use `github.com/stretchr/testify/assert`.
In this case, it would be:
assert.Error(t, err, "Expected an error with an invalid log level")